modules/apps/transfer/types/trace.go

```diff
@@ -71,6 +71,9 @@ func (dt DenomTrace) GetFullDenomPath() string {
 	if dt.Path == "" {
 		return dt.BaseDenom
 	}
+	if dt.BaseDenom == "" {
+		return dt.Path
+	}
```

Comment:
why was this added? I think in this situation, there should be a panic

Reply:
i added it bc in the case that the incorrectly parsed BaseDenom is an empty string for whatever reason, the `GetPrefix()` method will actually append an additional `/` onto the end and the BaseDenom will end up looking like ie `erc/0x00000000000etc/` at the conclusion of this logic. I can also set a panic, which do you think makes more sense?

Reply:
I think we can cleanup a lot of this code when we look into refactoring transfer and improving upon it. For now, I think it'd be best to remove this check and ensure during migrations that the previously set denom trace passes the `Validate()` call. If it doesn't, we should probably panic

Reply:
but i dont think some of the previous set denom traces would pass the Validate() call... ie something like ```transfertypes.DenomTrace{
BaseDenom: "1", Path: "transfer/channel-0/gamm/pool",
})```

wouldnt this make the point of the migration code redundant if we are trying to `Validate` before setting the new trace?

Reply:
previous invalid denom traces aren't possible in the current code as they would fail on packet [Validation](https://github.com/cosmos/ibc-go/blob/aaccbd193f19f97f8068278141044bf324e6425c/modules/apps/transfer/types/packet.go#L54). The purpose of the migration code is to fix the stored trace info, not validate the denomination. denomTrace.Validate() should pass for all existing denominations

The specific example, "gamm/pool/1", should have failed due to port/channel identifier validation (insufficient length) when being received or sent in a packet

Reply:
got it, that makes sense!